### PR TITLE
Use TravisCI, update documentation

### PR DESCRIPTION
**Pull request by jdenen**

### CODE REVIEW

File: lib/carpenter/application.ex [hunk 1]
```diff
@@ -1,14 +1,12 @@
 defmodule Carpenter.Application do
-  @moduledoc "false"
+  @moduledoc false
```
Comment (ManApart) [inline lib/carpenter/application.ex]: Should this have some actual documentation?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - chmod +x docker-compose
+  - sudo mv docker-compose /usr/local/bin
+install:
+  - mix local.rebar --force
+  - mix local.hex --force
+  - mix deps.get
+  - mix format --check-formatted
+  - mix credo
+script:
+  - docker build -t genesis:build .
```
Comment (jeffgrunewald) [inline .travis.yml]: copy pasta name

File: scripts/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+elif [[ $RELEASE_TYPE == "master" ]]; then
+    export TAGGED_IMAGE="smartcitiesdata/genesis:development"
+else
+    echo "Branch should not be pushed to Dockerhub"
+    exit 0
+fi
+
+echo "Pushing to Dockerhub with tag ${TAGGED_IMAGE} ..."
+
+docker tag genesis:build "${TAGGED_IMAGE}"
```
Comment (jeffgrunewald) [inline scripts/deploy.sh]: copy pasta name

File: scripts/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+RELEASE_TYPE=$1
+echo "Logging into Dockerhub ..."
+echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+
+echo "Determining image tag for ${TRAVIS_BRANCH} build ..."
+
+if [[ $RELEASE_TYPE == "release" ]]; then
+    export TAGGED_IMAGE="smartcitiesdata/genesis:${TRAVIS_BRANCH}"
```
Comment (jeffgrunewald) [inline scripts/deploy.sh]: copy pasta name

File: scripts/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+elif [[ $RELEASE_TYPE == "master" ]]; then
+    export TAGGED_IMAGE="smartcitiesdata/genesis:development"
```
Comment (jeffgrunewald) [inline scripts/deploy.sh]: copy pasta name

File: lib/carpenter/application.ex (hunk 1 above)
Comment (jdenen) [reply lib/carpenter/application.ex]: I don't think so. Especially since it's not a library.

File: .travis.yml (hunk 2 above)
Comment (jdenen) [reply .travis.yml]: nice catch

File: scripts/deploy.sh (hunk 5 above)
Comment (bbrewer-papa-x) [reply scripts/deploy.sh]: > copy pasta name

copy pasta name

**Pull request merged**